### PR TITLE
Added in issue and pr example tempates

### PR DESCRIPTION
**Pull request by brechmos-stsci**
I added in an example issue and PR template that could be added into a github repo. The PR template, in particular, follows the https://github.com/spacetelescope/style-guides/blob/master/guides/git-pr-review.md.

### CODE REVIEW

**Comment by pllim** [comment]
Just curious, did you generate the initial version with the "choose your own adventure" site that you shared?

File: templates/PULL_REQUEST_TEMPLATE [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+<!--- Please describe how you tested your changes. -->
+<!--- Include details of your testing environment, and the tests you ran to -->
+<!--- see how your change affects other areas of the code, etc. -->
+
+## Screenshots (if appropriate):
+
+## Checklist:
+
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
```
Comment (drdavella) [inline templates/PULL_REQUEST_TEMPLATE]: This checklist seems to be mostly for maintainers. Should we add a comment to clarify?

Comment (brechmos-stsci) [reply templates/PULL_REQUEST_TEMPLATE]: Hmm... good point. I'll add a comment to it.

Comment (brechmos-stsci) [reply templates/PULL_REQUEST_TEMPLATE]: @drdavella I actually removed the checklist section completely as it really isn't for the PR writer.  Does that resolve the requested change?

**Comment by brechmos-stsci** [comment]
@hcferguson Any comments?  
@drdavella You good with my changes?

**Comment by brechmos-stsci** [comment]
@stscicrawford feel free to comment too :)  I would like to start incorporating this into some repos when this is approved.

**Comment by hcferguson** [comment]
Looks fine. Not really a checklist, but that's okay. Do you want to make recommendations in the template about the scope of the PR -- that it is best to do small, focused PRs rather than collect a bunch of disparate changes into one big PR?

**Comment by brechmos-stsci** [comment]
@hcferguson I think the focus of the PR (ie, small) should be a discussion in the style guide as opposed to a point in the PR template. At the point that they have opened a PR, the work is probably already done and might be hard to backtrack.  (But, if you feel strongly about this, or others do, I can add in a comment like that.)

File: templates/PULL_REQUEST_TEMPLATE [hunk 2]
```diff
@@ -0,0 +1,15 @@
+## Description
+
+<!--- Describe your changes in detail -->
+
+Fixes: <!--- add github link -->
```
Comment (stscicrawford) [inline templates/PULL_REQUEST_TEMPLATE]: Does Github work with the ':' ?

File: templates/PULL_REQUEST_TEMPLATE [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+## How Has This Been Tested?
+
+<!--- Please describe how you tested your changes. -->
+<!--- Include details of your testing environment, and the tests you ran to -->
+<!--- see how your change affects other areas of the code, etc. -->
+
+## Screenshots (if appropriate):
```
Comment (stscicrawford) [inline templates/PULL_REQUEST_TEMPLATE]: I find that this is very visualization tools specific, but would not be appropriate in most cases.  As this is for the template, it is okay, but I might change to (if appropriate to something more descriptive so someone copying knows that they can remove it)

File: templates/PULL_REQUEST_TEMPLATE [hunk 4]
```diff
@@ -0,0 +1,15 @@
+## Description
+
+<!--- Describe your changes in detail -->
+
+Fixes: <!--- add github link -->
+
+## How Has This Been Tested?
+
+<!--- Please describe how you tested your changes. -->
```
Comment (stscicrawford) [inline templates/PULL_REQUEST_TEMPLATE]: [ ] Please describe how you tested your changes including your testing environment?
[ ] Have the unit tests for the package passed in your environment?
[ ] If appropriate, have the regression tests affected by your code change passed?

**Comment by stscicrawford** [comment]
Small comments on the PR template. 

My feeling is that the issue one could be more focused.  I think it would be confusing to someone that would be new and I would just have it focus on bug reports with the most important information first (what happened, traceback, and environment/versions)

I'm not sure if this is necessarily better, but this is the one we used in ccdproc as an example (this would need to be modified for STScI):


This is the template for bug reports, if you have a feature request or question
you can safely ignore and delete this prefilled text.

Include a description of the problem: What are you trying to do (include your
code and the **full** traceback)? What did you expect?

Include a minimal example to reproduce the issue including output and
traceback. The triple backticks make github render this as a multi-line code
block.

Don't forget to include the version of astropy, ccdproc and numpy, just copy
this into your Python interpreter (without the backticks):

```
import astropy
print(astropy.__version__)
import ccdproc
print(ccdproc.__version__)
import numpy
print(numpy.__version__)
```

**Comment by brechmos-stsci** [comment]
@stscicrawford I hear what you are saying but I think it would be helpful to prompt them for features too.   I think there are three ways this could be formatted:  1) aim for bug reports, primarily, as you suggested 2) reformat it a bit so it is similar to below where each section has the "if bug" / "if feature" type of thing, or 3) reformat so the "if bug" / "If feature" wraps the headings, text and prompts.

I am willing to go either way. I kind of like the reformatted below, but will reformat to a more focused bug report if you want.

```
## Describe the bug / feature

  <!--- If describing a bug:                                                          -->
  <!----   - tell us what happened instead of the expected behavior                   -->       
  <!---    - include the steps to reproduce the bug                                   -->       
  <!---    - if a Traceback is available, include the complete Traceback in the Issue -->       

  <!--- If a change/improvement/feature                                               -->
  <!---    - explain the idea and/or the difference from current behavior             -->

## Possible Solution

  <!--- Not obligatory, but suggest a fix/reason for the bug,                         -->       
  <!--- or ideas how to implement the addition or change (other code, article etc)    -->       

## Context

  <!--- Providing context helps us come up with a solution that is most useful in     -->
  <!--- the real world.                                                               -->       

  <!--- If a bug:                                                                     -->
  <!---    - What are you trying to accomplish?                                       -->       
  <!---    - If it is data specific, please provide any relevant information about    -->
  <!---      the data, or a link to an example, if possible.                          -->       

  <!--- If a new feature:                                                             -->       
  <!---    - How will this feature benefit the field?                                 -->       


## Your Environment

<!--- If a bug:                                                                       -->
<!---     include as many relevant details about the environment                      -->

  * Package Version used:
  * Operating System and version:
  * Include output of conda list:
```

**Comment by stscicrawford** [comment]
If you move the `environment information` into the first section under 'if a bug', then  I'd be fine with that.     That's an important bit of information so I think it would be good to include it in the first section so it isn't skipped or missed.

**Comment by brechmos-stsci** [comment]
@stscicrawford I moved the env info to the top part. I reworded a bit too.

**Comment by hcferguson** [comment]
@brechmos-stsci It's fine to put the comments about scope in a style guide rather than the template. Relevant checklist items might be:

- [ ]  Is this PR limited in scope?
- [ ] Are all the relevant issues tagged?

I think right now there is only a guide for reviewers, and I might not think to look there before submitting a PR. If you want to put all the info for submitters and reviewers in one guide, them maybe rename that "PR submitter and reviewer style guide?"

**Comment by brechmos-stsci** [comment]
@stscicrawford I saw you reference changes, just saw them now. Will do...

File: templates/PULL_REQUEST_TEMPLATE (hunk 2 above)
Comment (brechmos-stsci) [reply templates/PULL_REQUEST_TEMPLATE]: It does, there is an example (well for "Closes") in https://help.github.com/articles/closing-issues-using-keywords/

File: templates/PULL_REQUEST_TEMPLATE (hunk 4 above)
Comment (brechmos-stsci) [reply templates/PULL_REQUEST_TEMPLATE]: I think something half way in between might be better (??).  The first one asks for information and maybe isn't quite a checkbox yes/no thing.

```
<!--- Please describe how you tested your changes. -->                         

[ ] Have the unit tests for the package passed in your environment?            
[ ] If appropriate, have the regression tests affected by your code change passed?
```

File: templates/PULL_REQUEST_TEMPLATE (hunk 3 above)
Comment (brechmos-stsci) [reply templates/PULL_REQUEST_TEMPLATE]: I'll add a comment to say that the section can be deleted if not appropriate.

**Comment by brechmos-stsci** [comment]
@hcferguson I changed the text to the "fixes" part to say "all relevant issues" rather than just "the issue" to hopefully tweak people to put all the relevant ones.
The PR reviewer guide is specific to people reviewing PRs rather than writing PRs. There was some discussion that a "write a PR" style guide might be helpful but I think that is separate from "reviewing a PR" (not to say that reading the "reviewing a PR" might be helpful anyway!

**Pull request merged**